cli/cli
# Add handling of empty titles for Issues and PRs

Closes #9698 

## Background

Currently, the prompter doesn't enforce the requirement of a title for PR and Issue creation. However, that creates the UX where a user can enter a blank title, spend time filling out a detailed body, then fail to create the issue or pr when the request is sent to the api because the title is blank. This attempts to handle that before sending a request to the api, enforcing that, when prompting, the title of an issue is not blank.

## Design

This was a relatively easy change to make. I've included a screenshot of the current design below (it matches what was suggested in #9698):

<img width="630" alt="Screenshot 2024-10-08 at 4 33 59 PM" src="https://github.com/user-attachments/assets/3fd48f8e-8577-4fa4-a119-736cc940fabc">

### Re-prompting vs exiting

**Current UX:** _User inputs blank title_ - prompts for the rest of the issue info then fails when issue is created:
**UX in this PR:** _User inputs blank title_ - gives error message and re-prompts for title again

I chose to go with the re-prompting method - give the user an error message and prompt again - as I think it's better UX than just failing outright.

## Testing

1. Pull down branch
2. Build: `make`
3. Run `gh issue create`
4. When prompted for a title, press enter
5. Observe change
6. When prompted for a title again, enter a non-blank title
7. Complete flow
